### PR TITLE
Added event when scan is done and a bool property whether Any mesh sectors been scanned

### PR DESCRIPTION
**Pull request by Sacristan**
Added Delegate when scan is done and a bool property whether Any mesh sectors been scanned

Why i did this?
I needed an delegate event when scanning is done;
In my application i need to know if something has been scanned to show additional label

### CODE REVIEW

**Comment by mattfedocsg** [comment]
Good addition! I was doing the same thing at one point.

File: Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstanding.cs [hunk 1]
```diff
@@ -36,6 +36,9 @@ public enum ScanStates
         [Tooltip("Update period used after the scanning process is completed")]
         public float UpdatePeriod_AfterScanning = 4.0f;
 
+        public event OnScanDoneDelegate OnScanDone;
+        public delegate void OnScanDoneDelegate();
```
Comment (StephenHodgson) [inline Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstanding.cs]: Nit: I know the order really doesn't matter to the compiler but I think the delegate should be declared above the event.

File: Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstanding.cs [hunk 2]
```diff
@@ -256,6 +265,7 @@ private void Update_Scan(float deltaTime)
 
                 // Mark it
                 ScanState = ScanStates.Done;
+                OnScanDone.Invoke();
```
Comment (StephenHodgson) [inline Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstanding.cs]: if (OnScanDone!=null)
{
    OnScanDone.Invode();
}

File: Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstanding.cs [hunk 3]
```diff
@@ -47,6 +50,12 @@ public bool AllowSpatialUnderstanding
                 return true;
             }
         }
+
+        /// <summary>
+        /// Has any mesh sectors been scanned
+        /// </summary>
+        public bool HasMeshSectors { get { return UnderstandingCustomMesh.HasMeshSectors; } }
```
Comment (StephenHodgson) [inline Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstanding.cs]: if `UnderstandingCustomMesh.HasMeshSectors` is public why even add this property?

File: Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstanding.cs (hunk 1 above)
Comment (Sacristan) [inline Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstanding.cs]: Allright!

File: Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstanding.cs (hunk 3 above)
Comment (Sacristan) [inline Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstanding.cs]: previously it was private - so this was implemented. Merged! Will remove the property

File: Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstanding.cs (hunk 2 above)
Comment (Sacristan) [inline Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstanding.cs]: sure - rookie error

Comment (StephenHodgson) [inline Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstanding.cs]: No worries, essentially you're just checking to make sure there's subscribers before invoking the event.

**Comment by Sacristan** [comment]
Please review newest commit!

File: Assets/HoloToolkit-Examples/SpatialUnderstanding/SpatialUnderstanding-FeatureOverview/Scenes/SpatialUnderstandingExample.unity [hunk 4]
```diff
@@ -1728,7 +1729,7 @@ Prefab:
         type: 2}
       propertyPath: surfaceMaterial
       value: 
-      objectReference: {fileID: 2100000, guid: 6cec46d5b336e2a4b9b2e1d3e8e3754c, type: 2}
+      objectReference: {fileID: 2100000, guid: ddc2ff62e97ce6d41991784c3aa8e233, type: 2}
```
Comment (StephenHodgson) [inline Assets/HoloToolkit-Examples/SpatialUnderstanding/SpatialUnderstanding-FeatureOverview/Scenes/SpatialUnderstandingExample.unity]: What is the object reference we're changing here?

File: Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstanding.cs [hunk 5]
```diff
@@ -83,6 +78,11 @@ private set
                 if (ScanStateChanged != null)
                 {
                     ScanStateChanged();
+
+                    if (scanState == ScanStates.Done)
+                    {
+                        if (OnScanDone != null) OnScanDone.Invoke();
+                    }
```
Comment (StephenHodgson) [inline Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstanding.cs]: Why move here?

File: ProjectSettings/EditorBuildSettings.asset [hunk 6]
```diff
@@ -4,4 +4,6 @@
 EditorBuildSettings:
   m_ObjectHideFlags: 0
   serializedVersion: 2
-  m_Scenes: []
+  m_Scenes:
+  - enabled: 1
+    path: Assets/HoloToolkit-Examples/SpatialUnderstanding/SpatialUnderstanding-FeatureOverview/Scenes/SpatialUnderstandingExample.unity
```
Comment (StephenHodgson) [inline ProjectSettings/EditorBuildSettings.asset]: Please revert this file.

**Comment by Sacristan** [comment]
Removed previous commit and pushed new one.

**Comment by Sacristan** [comment]
Added public SpatialMappingObserver SurfaceObserver { get { return surfaceObserver; } } to SpatialMappingManager.cs

File: Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstandingCustomMesh.cs [hunk 7]
```diff
@@ -60,6 +61,11 @@ public bool DrawProcessedMesh
         }
 
         /// <summary>
+        /// Has any mesh sectors been scanned
```
Comment (thebanjomatic) [inline Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstandingCustomMesh.cs]: nit: Have any mesh sectors been scanned

**Pull request merged**